tests/test_migration.py

```diff
@@ -0,0 +1,237 @@
+                        "sgAssoc": json.dumps(
+                            self.create_signedassoc(
+                                "email", address, mxid, ts, 0, 99999999999
+                            )
+                        ),
+                    }
+                )
+            else:
+                # create some casefold-conflicting associations
+                address = "BOB%d@example.com" % (i - 10)
+                mxid = "@BOB%d:example.com" % (i - 10)
+                ts = 10000 * i
+                associations.append(
+                    {
+                        "medium": "email",
+                        "address": address,
+                        "lookup_hash": self.calculate_lookup(address),
+                        "mxid": mxid,
+                        "ts": ts,
+                        "not_before": 0,
+                        "not_after": 99999999999,
+                        "originServer": originServer,
+                        "originId": i,
+                        "sgAssoc": json.dumps(
+                            self.create_signedassoc(
+                                "email", address, mxid, ts, 0, 99999999999
+                            )
+                        ),
+                    }
+                )
+
+        # add all associations to db
+        cur = self.sydent.db.cursor()
+
+        cur.executemany(
+            "INSERT INTO global_threepid_associations "
+            "(medium, address, lookup_hash, mxid, ts, notBefore, notAfter, originServer, originId, sgAssoc) "
+            "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+            [
+                (
+                    assoc["medium"],
+                    assoc["address"],
+                    assoc["lookup_hash"],
+                    assoc["mxid"],
+                    assoc["ts"],
+                    assoc["not_before"],
+                    assoc["not_after"],
+                    assoc["originServer"],
+                    assoc["originId"],
+                    assoc["sgAssoc"],
+                )
+                for assoc in associations
+            ],
+        )
+
+        self.sydent.db.commit()
+
+    def test_migration_email(self):
+        with patch("sydent.util.emailutils.smtplib") as smtplib:
+            templateFile = self.sydent.get_branded_template(
+                "matrix-org",
+                "migration_template.eml",
+                ("email", "email.template"),
+            )
+            sendEmail(
+                self.sydent,
+                templateFile,
+                "bob@example.com",
+                {
+                    "mxid": "@bob:example.com",
+                    "subject_header_value": "MatrixID Deletion",
+                },
+            )
+            smtp = smtplib.SMTP.return_value
+            email_contents = smtp.sendmail.call_args[0][2].decode("utf-8")
+            self.assertIn("This is a notification", email_contents)
+
+    def test_local_db_migration(self):
```

Comment:
also test that the e-mails were sent? :)

Reply:
I changed it and I think I'm testing if the email is sent now...

Reply:
I think we want to check that `update_local_associations` (& global equivalent) are sending e-mails -- and who to (probably sufficient to test that one is being sent to a casefolded person and that no e-mail is being sent to someone who hasn't been affected).